resources/prosody-plugins/mod_muc_password_check.lua

```diff
@@ -0,0 +1,207 @@
+    module:log("debug","Sending response for room password validate: %s", inspect(PUT_response));
+
+    return PUT_response;
+
+end
+
+--- Handles request for retrieving the room participants details
+-- @param event the http event, holds the request query
+-- @return GET response, containing a json with participants details
+function handle_get_room_password (event)
+    module:log("info","Request for room password received: reqid %s", event.request.headers["request_id"])
+    if (not event.request.url.query) then
+        module:log("warn", "No query");
+        return 400;
+    end
+    local params = parse(event.request.url.query);
+    local room_name = params["room"];
+    local domain_name = params["domain"];
+    local subdomain = params["subdomain"];
+    local conference = params["conferenceFullName"];
+
+    local room_address;
+
+    if (not conference) and ((not room_name) or (not domain_name)) then
+        module:log("warn", "Missing param conference or room_name and domain_name");
+        return 400;
+    end
+
+    if conference then
+        room_address = room_jid_match_rewrite(conference)
+    else
+        room_address = jid.join(room_name, muc_domain_prefix.."."..domain_name);
+
+        if subdomain and subdomain ~= "" then
+            room_address = "["..subdomain.."]"..room_address;
+        end
+    end
+
+
+    -- verify access
+    local token = event.request.headers["authorization"]
+
+    if token and starts_with(token,'Bearer ') then
+        token = token:sub(8,#token)
+    end
+
+    if not verify_token(token, room_address) then
+        return 403;
+    end
+
+        local room = get_room_from_jid(room_address);
+
+    if room then
+        room_details = {};
+        room_details["conference"] = room_address;
+        room_details["passcodeProtected"] = room:get_password() ~= nil;
+        room_details["lobbyEnabled"] = room._data ~= nil and room._data.lobbyroom ~= nil;
+
+        local GET_response = {
+            headers = {
+                content_type = "application/json";
+            };
+            body = json.encode(room_details);
+        };
+        module:log("debug","Sending response for room password: %s", inspect(GET_response));
+
+        return GET_response;
+    end
+
+    -- default case, return 404
+    return 404;
+end
+
+-- process a host module directly if loaded or hooks to wait for its load
+function process_host_module(name, callback)
+    local function process_host(host)
+        if host == name then
+            callback(module:context(host), host);
+        end
+    end
+
+    if prosody.hosts[name] == nil then
+        module:log('debug', 'No host/component found, will wait for it: %s', name)
+
+        -- when a host or component is added
+        prosody.events.add_handler('host-activated', process_host);
+    else
+        process_host(name);
+    end
+end
+
+process_host_module(muc_domain_base, function(host_module, host)
```

Comment:
Would it make sense to add some sort of generic root route for these internal APIs?